packages/pretty-format/src/index.js

```diff
@@ -41,6 +41,27 @@ const errorToString = Error.prototype.toString;
 const regExpToString = RegExp.prototype.toString;
 const symbolToString = Symbol.prototype.toString;
 
+// Return whether val is equal to both global and window object.
+let noGlobalWindow;
+let theGlobalWindow;
+const isGlobalWindow = val => {
+  if (noGlobalWindow) {
+    return false;
+  }
+  if (theGlobalWindow === undefined) {
+    try {
+      /* global window */
+      noGlobalWindow = global !== window;
+      if (!noGlobalWindow) {
+        theGlobalWindow = window;
+      }
+    } catch (e) {
+      noGlobalWindow = true;
+    }
+  }
+  return val === theGlobalWindow;
+};
+
```

Comment:
why not just:
```js
const isGlobalWindow = val => {
  try {
    /* global window */
    return window === global;
  } catch (error) {
    return false;
  }
};
```

Reply:
In answer to your question, must return result of comparison to `val`

The complexity is in case some overhead to enclose `val === window` in `try-catch`, lazy evaluation of `window` at first attempt to serialize an object.

The more I think about it though, considering `global` seems like a mistake.

Reply:
Do you think I should do a set of perf measurements compared to simplest possible code?

Reply:
I was just referring to make it a bit simpler (I now see I forgot to include `val` in the equation :D). If you could do some local perf testing, that's always nice 👍